src/groupeddataframe/grouping.jl

```diff
@@ -203,15 +203,14 @@ Not meant to be constructed directly, see `groupby` abnd
 provided for a GroupApplied object.
 
 """
-type GroupApplied
+immutable GroupApplied{T<:AbstractDataFrame}
     gd::GroupedDataFrame
-    vals::Vector
+    vals::Vector{T}
 
-    function GroupApplied(gd, vals)
-        if length(gd) != length(vals)
-            error("GroupApplied requires keys and vals be of equal length.")
-        end
-        new(gd, vals)
+    @compat function (::Type{GroupApplied})(gd::GroupedDataFrame, vals::Vector)
+        length(gd) == length(vals) ||
+            throw(DimensionMismatch("GroupApplied requires keys and vals be of equal length."))
+        new{eltype(vals)}(gd, vals)
```

Comment:
I thought that type parameters create some overhead for dispatch. Here there are no inter-argument type constraints, so it's not necessary. In terms of type inference and the resulting code both versions should be equivalent. But I can change it, if using type parameters is the recommended way.